cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/default-service/AwsDefaults.service.tsx

```diff
@@ -1,9 +1,16 @@
 // App imports
-import { AWSVirtualMachine } from '../../../../../swagger-api';
+import { AwsService, AWSVirtualMachine, AWSAvailabilityZone } from '../../../../../swagger-api';
 import { AWSKeyPair } from '../../../../../swagger-api/models/AWSKeyPair';
 import { first } from '../../../../../shared/utilities/Array.util';
 import { getDefaultNodeTypes } from '../../../../../shared/constants/defaults/aws.defaults';
+import { AWS_NODE_PROFILE_NAMES } from '../../aws-mc-basic/AwsManagementClusterBasic.constants';
 
+export interface AvailabilityZoneInstance {
```

Comment:
As I understand it, the `AvailabilityZoneInstance` object is the payload object we're constructing, using `AWSAvailabilityZone.name` and `workerNodeType`. Looking ahead to the advanced flows, where the user can select these items for themselves (one at a time), I wonder if it would be easier to store these data separately in the store (rather than in this object), and then pull them together into this object at the time we create the payload?

Reply:
Rename the variable. Create object with the following structure:
```
{
  availability-zone-1: { id: us-west-a, name: us-west-a},
  availability-zone-1-work-node-type: t3.larg,
  availability-zone-1-public-subnet-id: '',
  availability-zone-1-private-subnet-id: '',
  availability-zone-2: { id: us-east-a, name: us-east-a},
  availability-zone-2-work-node-type: t3.larg,
  availability-zone-2-public-subnet-id: '',
  availability-zone-2-private-subnet-id: '',
}
```